Add Charge Description dimension (#221)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: supporting_content/dimensions/chargedescription.md

# Column: ChargeDescription

## Example provider mappings

Current resource types found or extracted from available data sets:

| Provider  | Data set                | Column                                            	|
| :-------- | :---------------------- | :-------------------------------------------------------|
| AWS       | CUR                     | lineItem/LineItemDescription 				|
| GCP       | BigQuery Billing Export | sku.description                                         |
| Microsoft | Cost details            | Not available      			                |

## Example usage scenarios

| Provider  | Data set                | Scenario                           | Value                    |
|:----------|:------------------------|:-----------------------------------|:-------------------------|
| AWS       | CUR                     | Not available                      | $0.00 per GB - US West (Oregon) data transfer from US West (Northern California) |
| GCP       | BigQuery Billing Export | Not available                      | Not Available            |
| Microsoft | Cost details            | via Cost export file		   | Not Available            |

## Discussion Topics

> - Should we explicitly define the minimum required information for this column? A list of FOCUS (Irena)
> - Are these enough? Region/location? Qty? (Larry)
> 	- What: lineItem/LineItemDescription
> 	- Where: Region
> 	- How (many): Qty

Reviewer: ```suggestion
### Minimum Information Requirements vs Open-Ended

We deliberated if we should be more specific and define the minimum required information for this column? 
- A list of FOCUS columns which we believe provide this high-level context without the need for additional discovery and thus must be included (concatenated) as part of the Charge description? (Irena)

- Are these enough? Region/location? Qty? (Larry)
  - What: lineItem/LineItemDescription
  - Where: Region
  - How (many): Qty
Considering a variable landscape across cloud, SaaS, etc. we decided to keep it open-ended.
```

# Column: ChargeDescription

## Example provider mappings

Current resource types found or extracted from available data sets:

| Provider  | Data set                | Column                                            	|
| :-------- | :---------------------- | :-------------------------------------------------------|
| AWS       | CUR                     | lineItem/LineItemDescription 				|
| GCP       | BigQuery Billing Export | sku.description                                         |

> | Microsoft | Cost details            | Not available      			                |

Reviewer 2: I think `MeterName` is as close as it gets in Azure: [link](https://azure.microsoft.com/en-us/updates/update-reports-for-improved-meter-names/)

We can call it out as not a precise match because it requires other columns' values to have sufficient context, but this is what Ternary uses for this column.

Reviewer 3: Actually, `ProductName` is the same thing as `sku.description`, but with a little more context. 
```suggestion
| Microsoft | Cost details            | ProductName      			                |
```

Author: Given that this dimension is meant to be a bit subjective, I don't think it makes sense to try to concoct a list of columns that tries to mimic AWS' or GCP's corresponding columns.  While MeterName and ProductName may contain important aspects of the entire line item, they likely won't encompass summary.  Thoughts?

Reviewer 2: I think it is ok to call out that they all feel like a good fit sometimes, and there is not one exact fit in the current data format. That would be representative of my experience as a data platform creator.

Reviewer 2: Just bumping here, I would be in favor of either setting this row to ProductName (accept Michael's suggestion), setting it to "ProductName and MeterName provide similar information", but i feel strongly it shouldn't be marked as 'unavailable'

## Example usage scenarios

| Provider  | Data set                | Scenario                           | Value                    |
|:----------|:------------------------|:-----------------------------------|:-------------------------|
| AWS       | CUR                     | Not available                      | $0.00 per GB - US West (Oregon) data transfer from US West (Northern California) |
| GCP       | BigQuery Billing Export | Not available                      | Not Available            |
| Microsoft | Cost details            | via Cost export file		   | Not Available            |

> ## Discussion Topics

Reviewer 2: Took a crack at this. Not sure how it works with the suggestion system, so be careful if accepting.

```suggestion
## Guiding Examples

Given that this is a free-form text field, it may be difficult for providers to understand how they should create ChargeDescriptions for their services. The following table grounds out the recommendations above with guiding examples of ChargeDescription values, whether they are recommended, and why or why not.

| Example Value | Recommended | Reason |
| ----------    | ----------- | ------ |
| X3 Compute Core in USA | Yes | Specific, concise, and may be comprehensible even without the  `ServiceName` |
| Class B API Requests | Yes | Specific and concise, even though it might only make sense when combined with the `ServiceName` |
| Foobarsoft OS Licensing Fee | Yes | Specific, concise, and may be comprehensible even without the  `ServiceName` |
| $0.005 for the first 1,000 API requests - Free tier<br/>$0.005 for the next 10,000 API requests<br/>$0.0025 per 1,000 requests thereafter | *No* | Includes pricing which is redundant with another column; has multiple descriptions for the same fundamental usage and has multiple skus for the same fundamental usage |
| Megacloud Compute Platform X3 instances with Intel Core i25. Compute Engine is an awesome service and way better than the others. Use it today and try our Y5 as well! | **No** | Not concise; is essentially marketing copy |
| R3_USAGE | *No* | Opaque SKU ID-type or provider-internal identifier that is not necessarily human readable |
| Usage | *No* | Not specific enough |

## Discussion Topics
```

### Minimum Information Requirements vs Open-Ended

We deliberated if we should be more specific and define the minimum required information for this column?

> - A list of FOCUS columns which we believe provide this high-level context without the need for additional discovery and thus must be included (concatenated) as part of the Charge description? (Irena)

Reviewer 2: I would venture that the charge description should be unambiguous once combined with the  product name / service description information.

Reviewer 3: I think we need to be explicit here at least at a conceptual level. I'm fine if we don't give the algorithm for concatenation, but we need to be clear for providers who don't do this today. If we aren't, it's going to be wildly inconsistent and ultimately useless. What happens when we get a paragraph description of marketing content here? Is that okay? I'm assuming not.

Author: But, we can't think of this from just a cloud perspective.  From the SaaS angle, this column is meant to act as the part of the charge that gives the most context around what is being billed.  Most of the time, from an invoice or billing/usage summary from SaaS products, these details don't exist as columns to concatenate.

Reviewer 2: I know why you would say that! 

But I want to fight a little harder for CSP applicability. Remember we're getting articles written about us because of the applicability to public cloud. We cannot and should not walk away from that..

I am happy to address the SaaS case additionally in this copy, but I think it's super important to make sure things are super clear for current and future cloud-provider-type services.

Reviewer 2: added suggestion above. thanks for prodding
